### ISSUE TITLE
Some problems with baseline PPO1

### ISSUE THREAD

**Issue by zamlz** [openai/baselines#258]
Hi everyone. I'm trying to use the baseline PPO1 and am training it on my own environment. Here is the issue that I am facing. Everything seems to be going fine and it seems to be learning the behavior that I want it to learn but when it gets the the optimizing phase before iteration 3 something typically happens. Here is a screenshot of what I mean.
![image](https://user-images.githubusercontent.com/10750027/35189196-7a4c7d2a-fdfa-11e7-9002-b78f1d795d03.png)
Secondly, I mean to say its not always the same error every time. Maybe another time I try running it and it gives me the following for the first error:
`
d:\sys\openai_baselines\baselines\common\mpi_adam.py:33:
RuntimeWarning: overflow encountered in multiply.
self.v = self.beta2*self.v + (1 - self.beta2)*(globalg*globalg)
`
Regardless of what happens, after the warnings, all my values become Nan, and my policy just because a constant value with no changes. Does anyone know what could be causing this? Thanks!

**Comment by olegklimov** [comment]
Hi @zamlz ! Did you check for NaNs that your env produces? Did you see NaNs with any env that I have on my computer (something from gym?).

**Comment by zamlz** [comment]
It was definitely something with my Env, but not sure what. But I managed to get it fixed by changed the way the logits variable in cnn_policy worked. It gets its value from the network, which is something that mlp_policy does not do. I changed it be similar to mlp_policy and everything works again.

**Comment by olegklimov** [comment]
OK let's close this.

**Comment by zamlz** [comment]
So I think I've figured whats happening more or less. I came across the same error in another environment. The environment was never producing any NaNs but only accumulated negative reward. It hadn't reached the condition for positive reward yet. The reward function is a continuous function and based on the agent's actions it either increases or decreases. It also starts in state which gives it negative reward. I believe that because of a lack of positive rewards (due to it being sparse), there is too much negative loss and it causes it to overflow somehow. I verified this by trying a different reward function that was more progressive in giving out positive reward. This didn't cause any stability problems, although the new reward function produces undesirable behavior. 

Is this the intended behavior? I was under the impression that as long as the reward function has a well defined gradient and rewards values themselves at each time-step were small, it should be able to find a policy to maximize it. Is this a problem with PPO or is it the implementation?
